packages/cryptography/src/Mnemonic.js

```diff
@@ -125,7 +165,78 @@ export default class Mnemonic {
             return this.toLegacyPrivateKey();
         }
 
-        return this._toPrivateKey(passphrase);
+        let { keyData, chainCode } = await this._toKeyData(
+            passphrase,
+            ED25519_SEED_TEXT
+        );
+
+        for (const index of path) {
+            ({ keyData, chainCode } = await slip10.derive(
+                keyData,
+                chainCode,
+                index
+            ));
+        }
+
+        const keyPair = nacl.sign.keyPair.fromSeed(keyData);
+
+        if (CACHE.privateKeyConstructor == null) {
```

Comment:
nit: could use another internal method here and below
```suggestion
        return cacheNewPrivateKey(new Ed25519PrivateKey(keyPair, chainCode))
```

Where
```
cacheNewPrivateKey(privateKey) {
        if (CACHE.privateKeyConstructor == null) {
            throw new Error("PrivateKey not found in cache");
        }

        return CACHE.privateKeyConstructor(privateKey);
    }
```

Reply:
Although I do agree with this change, I would rather do this in another PR as I believe I could instead update `CACHE` to be a class with methods per field instead of making a singular method just for `Cache.privateKeyConstructor`
Created tracking issue: https://github.com/hashgraph/hedera-sdk-js/issues/1225